include/tins/cxxstd.h

```diff
@@ -31,6 +31,7 @@
 #define TINS_CXXSTD_H
 
 #include "config.h"
+#include <functional>
```

Comment:
Why is this here? Isn't this trying to fix something on `ip_address.h`?

Reply:
We believe we've found a report of a bug with clang that explains the error:
https://github.com/martinmoene/gsl-lite/issues/63

Originally, `functional` was included in `cxxstd.h` because `cxxstd.h` was included in the others.  Commit b0013d2 now attempts to use workaround in the issue above by putting `#include <functional>` at the end of `ipaddress.h` and `ipv6_address.h`.